vsix/CppWinRTPackageTemplateWizard.cs

```diff
@@ -0,0 +1,128 @@
+
+                Project msbuildProject = new Project(project.FullName);
+                if (msbuildProject.GetPropertyValue("CppWinRTDisableAutoNuGetReference").Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Don't forward this call to the NuGet wizard, as it has been explicitly disabled.
+                }
+                else
+                {
+                    wizardImpl?.ProjectFinishedGenerating(project);
+                }
+
+                ProjectCollection.GlobalProjectCollection.UnloadProject(msbuildProject);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void ProjectItemFinishedGenerating(DTEProjectItem projectItem)
+        {
+            VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
+            try
+            {
+                Project msbuildProject = new Project(projectItem.ContainingProject.FullName);
+                if (msbuildProject.GetPropertyValue("CppWinRTDisableAutoNuGetReference").Equals("true", StringComparison.OrdinalIgnoreCase))
+                {
+                    // Don't forward this call to the NuGet wizard, as it has been explicitly disabled.
+                    // NuGet packages are also installed in ProjectItemFinishedGenerating().
+                }
+                else
+                {
+                    wizardImpl?.ProjectItemFinishedGenerating(projectItem);
+                }
+
+                ProjectCollection.GlobalProjectCollection.UnloadProject(msbuildProject);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void RunFinished()
+        {
+            try
+            {
+                wizardImpl?.RunFinished();
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public void RunStarted(object automationObject, Dictionary<string, string> replacementsDictionary, WizardRunKind runKind, object[] customParams)
+        {
+            try
+            {
+                Assembly asm = Assembly.Load("NuGet.VisualStudio.Interop, Version=1.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a");
+                Type wizardType = asm.GetType("NuGet.VisualStudio.TemplateWizard");
+
+                wizardImpl = (IWizard)Activator.CreateInstance(wizardType);
+                wizardImpl.RunStarted(automationObject, replacementsDictionary, runKind, customParams);
+            }
+            catch (Exception ex)
+            {
+                ShowExceptionDialog(ex);
+            }
+        }
+
+        public bool ShouldAddProjectItem(string filePath)
+        {
+            try
+            {
+                if (wizardImpl != null) return wizardImpl.ShouldAddProjectItem(filePath);
+                else return true;
```

Comment:
```suggestion
                return wizardImpl == null || wizardImpl.ShouldAddProjectItem(filePath);
```

Reply:
I don't like this idea; I think this style (conditional on left, method call or `throw` on the right) is much less readable.

Reply:
Not a problem since both variants will be jitted identically, but let's put under if-else. This style even is used in C++ files here:

```
if (condition)
    return call();
else
    return true;
```

Reply:
Or even better make `return true` the last line in the method. This will allow you to remove needless `throw` from the `catch` block.